### PR TITLE
update recommendation for async check

### PR DESCRIPTION
**Pull request by olegbespalov**
## What?

This change updates the recommendation for using the check polyfill that does support async functions. 

https://grafana.com/docs/k6/latest/javascript-api/jslib/utils/check/

## Why?

It improves UX of k6.

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [x] I have run linter locally (`make lint`) and all checks pass.
- [x] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Review by oleiade — APPROVED** [review]
👌🏻

**Review by joanlopez — APPROVED** [review]

File: js/modules/k6/k6.go [hunk 1]
```diff
@@ -182,7 +179,9 @@ func (mi *K6) Check(arg0, checks sobek.Value, extras ...sobek.Value) (bool, erro
 		}
 
 		if common.IsAsyncFunction(rt, val) {
-			return false, fmt.Errorf(asyncFunctionNotSupportedMsg, "check")
+			return false, errors.New("built-in check() does not support async functions as arguments, " +
```
Comment (mstoykov) [inline js/modules/k6/k6.go:182]: It is probably good idea either here or/and the linked utils to explain why this is a problem similar to how it does it fro group.

**Review by mstoykov — APPROVED** [review]

File: js/modules/k6/k6.go [hunk 2]
```diff
@@ -104,7 +100,8 @@ func (mi *K6) Group(name string, val sobek.Value) (sobek.Value, error) {
 		return nil, errors.New("group() requires a callback as a second argument")
 	}
 	if common.IsAsyncFunction(mi.vu.Runtime(), val) {
-		return sobek.Undefined(), fmt.Errorf(asyncFunctionNotSupportedMsg, "group")
+		return sobek.Undefined(), errors.New("group() does not support async functions as arguments, " +
+			"please see https://grafana.com/docs/k6/latest/javascript-api/k6/group/#working-with-async-functions for more info")
```
Comment (heitortsergent) [inline js/modules/k6/k6.go:104]: Just as a small nit, we don't have the heading #working-with-async-functions on that page, so you could just link to: https://grafana.com/docs/k6/latest/javascript-api/k6/group/.

File: js/modules/k6/k6.go (hunk 1 above)
Comment (heitortsergent) [inline js/modules/k6/k6.go:182]: "The built-in check() does not support async functions as arguments. Use the JavaScript utils library as a replacement. Refer to https://grafana.com/docs/k6/latest/javascript-api/jslib/utils/check/ for more info" 

Small suggestion for the message on these lines.

**Review by heitortsergent — COMMENTED** [review]
I know this already got merged, but just left two small suggestions for updating the messages. 🤓

File: js/modules/k6/k6.go (hunk 2 above)
Comment (joanlopez) [reply js/modules/k6/k6.go:104]: Good catch, thanks! @heitortsergent

File: js/modules/k6/k6.go (hunk 1 above)
Comment (olegbespalov) [reply js/modules/k6/k6.go:182]: Thanks! I've opened the https://github.com/grafana/k6/pull/4043 for addressing :+1:

Comment (mstoykov) [reply js/modules/k6/k6.go:182]: @heitortsergent maybe we should fix this part in the docs ?

Comment (heitortsergent) [reply js/modules/k6/k6.go:182]: Yeah, that makes sense @mstoykov. I'll open an issue for it. 🙇

**Pull request merged**